### PR TITLE
Change Rosetta API to show general operation status

### LINKED ISSUE #5328: Rosetta show general operation status for new HAPI response codes
### Problem

As a rosetta user, I want the DATA API to show general operation status for any new hedera HAPI response codes unless the new response code indicates a success operation or needs special handling than any other non-success response code. 

### Solution

- Add a general status `GENERAL_ERROR`
- For any new HAPI response codes not in `types.TransactionResults`, set the corresponding rosetta operation's status to the general one

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by mgoelswirlds**
**Description**:
Adding a general operational status to accomodate HAPI response codes not handled.

This PR modifies ... in order to support ...
* Add GENERAL_ERROR status to support HAPI response codes not handles in TransactionResults.

**Related issue(s)**:

Fixes #5328

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit)

### CODE REVIEW

File: hedera-mirror-rosetta/app/domain/types/constants.go [hunk 1]
```diff
@@ -23,6 +23,7 @@ package types
 import "github.com/coinbase/rosetta-sdk-go/types"
 
 const (
+	GeneralStatus   = "GENERAL_ERROR"
```
Comment (xin-hedera) [inline hedera-mirror-rosetta/app/domain/types/constants.go:26]:
```suggestion
	generalErrorStatus   = "GENERAL_ERROR"
```

symbols starting with uppercase letter is public in go, we don't want to make `generalStatus` public

also change `TransactionResults` to private,  i.e., `transactionResults`. then add the function `GetTransactionResult` here instead in `persistence/transaction.go`

File: hedera-mirror-rosetta/app/persistence/transaction.go [hunk 2]
```diff
@@ -498,6 +500,15 @@ func (tr *transactionRepository) constructTransaction(sameHashTransactions []*tr
 	return result, nil
 }
 
+func getTransactionResult(transaction *transaction) interface{} {
```
Comment (xin-hedera) [inline hedera-mirror-rosetta/app/persistence/transaction.go:503]: this function should move to types/constants.go, and the prototype should be
```suggestion
func GetTransactionResult(int32 code) string {
```

we know it always returns a `string`, so shouldn't use the generic empty `interface{}`

**Review by xin-hedera — CHANGES_REQUESTED** [review]

**Comment by xin-hedera** [comment]
please merge changes from `main` for the hedera-protobufs-go bump and new response codes

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**